README.md

```diff
@@ -1 +1,174 @@
-# new
\ No newline at end of file
+# Automated-Essay-Scoring: Model Deployment and Monitoring on Kubernetes
+## Contents
+- [Introduction](#introduction)
+- [API Deployment on Azure Kubernetes](#api-deployment-on-azure-kubernetes)
+- [CI/CD Pipeline with Jenkins](#cicd-pipeline-with-jenkins)
+- [Monitoring](#monitoring)
+
+## Introduction
+
+This project is an NLP-based Automated Essay Scoring system that leverages various cutting-edge technologies to provide a comprehensive solution for automatic grading. The key components and technologies used in this project include:
+
+- **Git/GitHub**: Version control and collaboration.
+- **Modeling**: Utilizes Hugging Face and Kaggle for training and fine-tuning NLP models.
+- **Docker**: Containerization of the application for consistency across environments.
+- **FastAPI**: A high-performance API framework used for building the essay scoring API.
+- **Locust**: Load testing tool to ensure the API can handle high traffic.
+- **Kubernetes (K8s) & Helm**: Service deployment and management, with NGINX Ingress for routing traffic.
+- **CI/CD with Jenkins**: Automates the build, test, and deployment pipeline to streamline development.
+- **Monitoring**: Integrates Elasticsearch, Filebeat, Kibana, Prometheus, and Grafana for logging, metrics collection, and visualization.
+- **Cloud Platform**: Microsoft Azure for deploying and managing cloud infrastructure.
+- **Infrastructure as Code**: Ansible and Terraform are used for provisioning and managing infrastructure.
+- **Pre-trained Model**: Microsoft DeBERTa v3 small model serves as the foundation for essay scoring.
+
+### Project Structure
+```bash
+Automated-Essay-Scoring
+    ├── deployments
+    │   ├── ess
+    │   └── nginx-ingress
+    ├── Dockerfile
+    ├── elk
+    │   ├── elastic-elasticsearch
+    │   ├── elastic-filebeat
+    │   └── elastic-kibana
+    ├── iac
+    │   ├── ansible
+    │   └── terraform
+    ├── jenkins
+    │   ├── docker-compose.yaml
+    │   ├── Dockerfile
+    │   └── README.md
+    ├── Jenkinsfile
+    ├── locustfile.py
+    ├── main.py
+    ├── models
+    │   └── model_weighted_training
+    ├── notebooks
+    │   ├── notebook_normal_tranning.ipynb
+    │   └── notebook_weighted_tranning.ipynb
+    ├── prometheus
+    │   ├── kube-prometheus-stack
+    │   ├── kube-prometheus-stack-62.3.1.tgz
+    │   └── values-prometheus.yaml
+    ├── README.md
+    ├── request.py
+    ├── requirements.txt
+    └── tests
+        └── test_model_correctness.py
+```
+
+![Diagram](images/flow_diagram.png)
+
+## API Deployment on Azure Kubernetes
+### 1 Create Azure Kubernetes Service using Terrafrom
+#### 1.1 Install Azure CLI and Login to Azure 
```

Comment:
=> #### 1.1. Install Azure CLI and Login to Azure, tương tự như mấy ô dưới nhé, để dấu . sau số